dolphinscheduler-task-plugin/dolphinscheduler-task-sql/src/main/java/org/apache/dolphinscheduler/plugin/task/sql/SqlTask.java

```diff
@@ -506,7 +506,7 @@ private SqlBinds getSqlAndSqlParamsMap(String sql) {
         // replace the ${} of the SQL statement with the Placeholder
         String formatSql = sql.replaceAll(rgex, "?");
         // Convert the list parameter
-        formatSql = ParameterUtils.expandListParameter(sqlParamsMap, formatSql);
+        formatSql = ParameterUtils.expandListParameter(sqlParamsMap, sql, rgex);
```

Comment:
i think its better to merge `sql.replaceAll` and `ParameterUtils.expandListParameter`, only one replace like `replaceOriginalValue`